core/common/src/main/java/alluxio/security/authentication/ChannelAuthenticator.java

```diff
@@ -181,4 +149,77 @@ public Channel authenticate(ManagedChannel managedChannel, AlluxioConfiguration
     }
     return interceptorsList;
   }
+
+  private class AuthenticatedManagedChannel extends Channel implements AuthenticatedChannel {
+    private final ManagedChannel mManagedChannel;
+    private final AlluxioConfiguration mConf;
+    private Channel mChannel;
+    private boolean mAuthenticated;
+
+    AuthenticatedManagedChannel(ManagedChannel managedChannel, AlluxioConfiguration conf) {
+      mManagedChannel = managedChannel;
+      mConf = conf;
+    }
+
+    void authenticate() throws AlluxioStatusException {
+      try {
+        // Create a channel for talking with target host's authentication service.
+        // Create SaslClient for authentication based on provided credentials.
+        SaslClient saslClient;
+        if (mUseSubject) {
+          saslClient = SaslParticipantProvider.Factory.create(mAuthType)
+              .createSaslClient(mParentSubject, mConf);
+        } else {
+          saslClient = SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mUserName,
+              mPassword, mImpersonationUser);
+        }
+
+        // Create authentication scheme specific handshake handler.
+        SaslHandshakeClientHandler handshakeClient =
+            SaslHandshakeClientHandler.Factory.create(mAuthType, saslClient);
+        // Create driver for driving sasl traffic from client side.
+        SaslStreamClientDriver clientDriver =
+            new SaslStreamClientDriver(handshakeClient, mGrpcAuthTimeoutMs);
+        // Start authentication call with the service and update the client driver.
+        StreamObserver<SaslMessage> requestObserver =
+            SaslAuthenticationServiceGrpc.newStub(mManagedChannel).authenticate(clientDriver);
+        clientDriver.setServerObserver(requestObserver);
+        // Start authentication traffic with the target.
+        clientDriver.start(mChannelId.toString());
+        // Authentication succeeded!
+        mAuthenticated = true;
+        mManagedChannel.notifyWhenStateChanged(ConnectivityState.READY, () -> {
+          mAuthenticated = false;
+        });
```

Comment:
Does this mean we will flag the channel if it goes into an `IDLE` state?

Reply:
yep. In gRPC `IDLE` state means that the channel does not have an active connection, thus here we reset the `mAuthenticated` flag to notify client pool to drop the channel instead of reusing it without reauthentication.